libc/stdio_scanf.c

```diff
@@ -0,0 +1,4291 @@
+	RUN_TEST_CASE(stdio_scanf_i, li);
+	RUN_TEST_CASE(stdio_scanf_i, lli);
+	RUN_TEST_CASE(stdio_scanf_i, ji);
+	RUN_TEST_CASE(stdio_scanf_i, zi);
+	RUN_TEST_CASE(stdio_scanf_i, ti);
+	remove(TESTFILE_PATH);
+}
+
+
+TEST_GROUP_RUNNER(stdio_scanf_u)
+{
+	RUN_TEST_CASE(stdio_scanf_u, u);
+	RUN_TEST_CASE(stdio_scanf_u, hhu);
+	RUN_TEST_CASE(stdio_scanf_u, hu);
+	RUN_TEST_CASE(stdio_scanf_u, lu);
+	RUN_TEST_CASE(stdio_scanf_u, llu);
+	RUN_TEST_CASE(stdio_scanf_u, ju);
+	RUN_TEST_CASE(stdio_scanf_u, zu);
+	RUN_TEST_CASE(stdio_scanf_u, tu);
+	remove(TESTFILE_PATH);
+}
+
+
+TEST_GROUP_RUNNER(stdio_scanf_o)
+{
+	RUN_TEST_CASE(stdio_scanf_o, o);
+	RUN_TEST_CASE(stdio_scanf_o, hho);
+	RUN_TEST_CASE(stdio_scanf_o, ho);
+	RUN_TEST_CASE(stdio_scanf_o, lo);
+	RUN_TEST_CASE(stdio_scanf_o, llo);
+	RUN_TEST_CASE(stdio_scanf_o, jo);
+	RUN_TEST_CASE(stdio_scanf_o, zo);
+	RUN_TEST_CASE(stdio_scanf_o, to);
+	remove(TESTFILE_PATH);
+}
+
+
+TEST_GROUP_RUNNER(stdio_scanf_x)
+{
+	RUN_TEST_CASE(stdio_scanf_x, x);
+	RUN_TEST_CASE(stdio_scanf_x, hhx);
+	RUN_TEST_CASE(stdio_scanf_x, hx);
+	RUN_TEST_CASE(stdio_scanf_x, lx);
+	RUN_TEST_CASE(stdio_scanf_x, llx);
+	RUN_TEST_CASE(stdio_scanf_x, jx);
+	RUN_TEST_CASE(stdio_scanf_x, zx);
+	RUN_TEST_CASE(stdio_scanf_x, tx);
+	RUN_TEST_CASE(stdio_scanf_x, X);
+	RUN_TEST_CASE(stdio_scanf_x, hhX);
+	RUN_TEST_CASE(stdio_scanf_x, hX);
+	RUN_TEST_CASE(stdio_scanf_x, lX);
+	RUN_TEST_CASE(stdio_scanf_x, llX);
+	RUN_TEST_CASE(stdio_scanf_x, jX);
+	RUN_TEST_CASE(stdio_scanf_x, zX);
+	RUN_TEST_CASE(stdio_scanf_x, tX);
+	remove(TESTFILE_PATH);
+}
+
+
+TEST_GROUP_RUNNER(stdio_scanf_aefg)
+{
+	RUN_TEST_CASE(stdio_scanf_aefg, f);
+	RUN_TEST_CASE(stdio_scanf_aefg, F);
+	RUN_TEST_CASE(stdio_scanf_aefg, a);
+	RUN_TEST_CASE(stdio_scanf_aefg, A);
+	RUN_TEST_CASE(stdio_scanf_aefg, e);
+	RUN_TEST_CASE(stdio_scanf_aefg, E);
+	RUN_TEST_CASE(stdio_scanf_aefg, g);
+	RUN_TEST_CASE(stdio_scanf_aefg, G);
+	RUN_TEST_CASE(stdio_scanf_aefg, inf_nan_f);
+	RUN_TEST_CASE(stdio_scanf_aefg, inf_nan_a);
+	RUN_TEST_CASE(stdio_scanf_aefg, inf_nan_e);
+	RUN_TEST_CASE(stdio_scanf_aefg, inf_nan_g);
+	remove(TESTFILE_PATH);
+}
+
+
+TEST_GROUP_RUNNER(stdio_scanf_cspn)
+{
+	RUN_TEST_CASE(stdio_scanf_cspn, c);
+	RUN_TEST_CASE(stdio_scanf_cspn, c_ascii);
+	RUN_TEST_CASE(stdio_scanf_cspn, s_path);
+	RUN_TEST_CASE(stdio_scanf_cspn, s_torn);
+	RUN_TEST_CASE(stdio_scanf_cspn, s_ascii);
+	RUN_TEST_CASE(stdio_scanf_cspn, s_pick);
+	RUN_TEST_CASE(stdio_scanf_cspn, percent);
+	RUN_TEST_CASE(stdio_scanf_cspn, n);
+	RUN_TEST_CASE(stdio_scanf_cspn, ptr);
+	remove(TESTFILE_PATH);
+}
+
+
+TEST_GROUP_RUNNER(stdio_scanf_squareBrackets)
+{
+	RUN_TEST_CASE(stdio_scanf_squareBrackets, simple);
+	RUN_TEST_CASE(stdio_scanf_squareBrackets, crircumflex);
+	RUN_TEST_CASE(stdio_scanf_squareBrackets, pos);
+	RUN_TEST_CASE(stdio_scanf_squareBrackets, white_spaces);
+	RUN_TEST_CASE(stdio_scanf_squareBrackets, ascii);
+	remove(TESTFILE_PATH);
+}
+
+
+TEST_GROUP_RUNNER(stdio_scanf_rest)
```

Comment:
Please also add a test case that checks different `optional non-zero decimal integers` before a few specifiers, please focus on edge cases. For example getting "%10s", when a string contains 11 elements, various cases with integers, floating point numbers